src/Swarm/Game/Entity.hs

```diff
@@ -523,18 +523,38 @@ fromList = foldl' (flip insert) empty
 
 -- | Create an inventory from a list of entities and their counts.
 fromElems :: [(Count, Entity)] -> Inventory
-fromElems = foldl' (flip (uncurry insertCount)) empty
+fromElems = insertInventoryList empty
+
+insertInventoryList :: Inventory -> [(Count, Entity)] -> Inventory
+insertInventoryList = foldl' (flip (uncurry insertCount))
+
+-- | Internal way to shift numbers to positive integers when hashing.
+--
+-- THIS DOES NOT FOLLOW ARITHMETIC LAWS, USE EXTREME CAUTION!
+hashCount :: Num p => Number -> p
+hashCount c = case c of
+  PosInfinity -> 1
+  Integer a | a >= 0 -> 1 + fromIntegral a
```

Comment:
Doesn't this mean `PosInfinity` and `Integer 0` both turn into `1`?  Maybe I'm not understanding what this function is for.

Reply:
Ah, I think I originally had `2 +` and then mistakenly changed that. That will be a pain to adjust again. :slightly_frowning_face:

Reply:
To be clear, I don't yet fully understand how this is used so it might be correct.